c/sedona-geos/src/st_buffer.rs

```diff
@@ -115,6 +151,115 @@ fn invoke_scalar(
     Ok(())
 }
 
+fn extract_optional_string(arg: Option<&ColumnarValue>) -> Result<Option<String>> {
+    let Some(arg) = arg else { return Ok(None) };
+    let casted = arg.cast_to(&DataType::Utf8, None)?;
+    match &casted {
+        ColumnarValue::Scalar(ScalarValue::Utf8(Some(s)) | ScalarValue::LargeUtf8(Some(s))) => {
+            Ok(Some(s.clone()))
+        }
+        ColumnarValue::Scalar(scalar) if scalar.is_null() => Ok(None),
+        ColumnarValue::Scalar(_) => Ok(None),
+        _ => Err(DataFusionError::Execution(format!(
+            "Expected scalar bufferStyleParameters, got: {:?}",
+            arg
+        ))),
+    }
+}
+
+fn parse_buffer_params(params_str: Option<&str>) -> Result<BufferParams> {
+    let Some(params_str) = params_str else {
+        return BufferParams::builder()
+            .build()
+            .map_err(|e| DataFusionError::External(Box::new(e)));
+    };
+
+    let mut params_builder = BufferParams::builder();
+
+    for param in params_str.split_whitespace() {
+        let Some((key, value)) = param.split_once('=') else {
+            return Err(DataFusionError::Execution(format!(
+                "Missing value for buffer parameter: {}",
+                param
+            )));
+        };
+
+        match key.to_lowercase().as_str() {
+            "endcap" => {
+                params_builder = params_builder.end_cap_style(parse_cap_style(value)?);
+            }
+            "join" => {
+                params_builder = params_builder.join_style(parse_join_style(value)?);
+            }
+            "side" => {
+                params_builder = params_builder.single_sided(parse_side(value)?);
+            }
+            "mitre_limit" | "miter_limit" => {
+                let limit: f64 = parse_number(value, "mitre_limit")?;
+                params_builder = params_builder.mitre_limit(limit);
+            }
+            "quad_segs" | "quadrant_segments" => {
+                let segs = parse_number(value, "quadrant_segments")?;
+                params_builder = params_builder.quadrant_segments(segs);
+            }
+            _ => {
+                return Err(DataFusionError::Execution(format!(
+                    "Invalid buffer parameter: {} (accept: 'endcap', 'join', 'mitre_limit', 'miter_limit', 'quad_segs' and 'side')",
+                    key.to_lowercase()
+                )));
+            }
+        }
+    }
+
+    params_builder
+        .build()
+        .map_err(|e| DataFusionError::External(Box::new(e)))
+}
+
+fn parse_cap_style(value: &str) -> Result<CapStyle> {
+    match value.to_lowercase().as_str() {
```

Comment:
nit: this will cause allocations.
You may want to use eq_ignore_ascii_case() instead.